tests/rptest/tests/read_replica_e2e_test.py

```diff
@@ -144,38 +143,17 @@ def test_simple_end_to_end(self, partition_count, min_records):
         rpk = RpkTool(self.redpanda)
         rpk.alter_topic_config(spec.name, 'redpanda.remote.write', 'true')
 
-        # Make sure all produced data is uploaded to S3
-        def s3_has_all_data():
-            objects = list(
-                self.redpanda._s3client.list_objects(self.s3_bucket_name))
-            total_uploaded = 0
-            for o in objects:
-                if o.Key.endswith(
-                        "/manifest.json") and self.topic_name in o.Key:
-                    data = self.redpanda._s3client.get_object_data(
-                        self.s3_bucket_name, o.Key)
-                    manifest = json.loads(data)
-                    last_upl_offset = manifest['last_offset']
-                    total_uploaded += last_upl_offset
-                    self.logger.info(
-                        f"Found manifest at {o.Key}, last_offset is {last_upl_offset}"
-                    )
-            self.logger.info(
-                f"Total uploaded: {total_uploaded}, num_acked: {self.producer.num_acked}"
-            )
-            return total_uploaded >= self.producer.num_acked
-
+        # Start second cluster and create read replica topic
+        self.start_second_cluster()
         wait_until(
-            s3_has_all_data,
+            self.create_read_replica_topic_success,
             timeout_sec=
-            30,  #should be uploaded since cloud_storage_segment_max_upload_interval_sec=5
+            300,  #should be uploaded since cloud_storage_segment_max_upload_interval_sec=5
             backoff_sec=5,
             err_msg=
-            f"Not all data is uploaded to S3 bucket, is S3 bucket: {list(self.redpanda._s3client.list_objects(self.s3_bucket_name))}"
+            f"Not all data is uploaded to S3 bucket, is S3 bucket: {list(self.redpanda._s3client.list_objects(self.si_settings.cloud_storage_bucket))}"
```

Comment:
This string is getting formatted when we enter wait_until, right?  I think you'd rather see the bucket listing at the point of the timeout, rather than before we started

Reply:
that's an error message that is printed in case of timeout

Reply:
Yes, but in the time sequence we're doing:
1. Process the format string, including listing objects in the bucket, and assign the result to `err_msg`
2. Attempt A...
3. Attempt B...
4. Attempt C...
5. On timeout, print `err_msg`.  But this is the list of objects from time 1, not from now.

Reply:
good catch! deleted s3 content from the error message. The s3 content is printed on deletion of s3 bucket, so it's still possible to get this information